src/arrayMethodSort.js

```diff
@@ -1,11 +1,32 @@
+/* eslint-disable no-param-reassign */
 'use strict';
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (typeof compareFunction !== 'function') {
+      compareFunction = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        return strA < strB ? -1 : strA > strB ? 1 : 0;
```

Comment:
Avoid nested ternary operators

Reply:
OK, but why? It's not so hard to understand the logic behind this strip of code.